src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -531,19 +526,28 @@ public ActivityProfileState isActive( final AlignmentContext context, final Refe
         }
 
         if (assemblyDebugOutStream != null) {
-            assemblyDebugOutStream.println("\n\n\n\n"+region.getSpan()+"\nNumber of reads in region: " + region.getReads().size() + "     they are:");
-            for (GATKRead read : region.getReads()) {
-                assemblyDebugOutStream.println(read.getName() + "   " + read.convertToSAMRecord(region.getHeader()).getFlags());
+            try {
+                assemblyDebugOutStream.write("\n\n\n\n" + region.getSpan() + "\nNumber of reads in region: " + region.getReads().size() + "     they are:");
+                for (GATKRead read : region.getReads()) {
+                    assemblyDebugOutStream.write(read.getName() + "   " + read.convertToSAMRecord(region.getHeader()).getFlags());
+                }
+            } catch (IOException e) {
+                throw new UserException("Error writing to debug output stream", e);
             }
-        }
+    }
 
-        // run the local assembler, getting back a collection of information on how we should proceed
+
+    // run the local assembler, getting back a collection of information on how we should proceed
         final AssemblyResultSet untrimmedAssemblyResult =  AssemblyBasedCallerUtils.assembleReads(region, givenAlleles, hcArgs, readsHeader, samplesList, logger, referenceReader, assemblyEngine, aligner, !hcArgs.doNotCorrectOverlappingBaseQualities);
 
         if (assemblyDebugOutStream != null) {
-            assemblyDebugOutStream.println("\nThere were " + untrimmedAssemblyResult.getHaplotypeList().size() + " haplotypes found. Here they are:");
-            for (String haplotype : untrimmedAssemblyResult.getHaplotypeList().stream().map(haplotype -> haplotype.toString()).sorted().collect(Collectors.toList())) {
-                assemblyDebugOutStream.println(haplotype);
+            try {
+                assemblyDebugOutStream.write("\nThere were " + untrimmedAssemblyResult.getHaplotypeList().size() + " haplotypes found. Here they are:");
+                for (String haplotype : untrimmedAssemblyResult.getHaplotypeList().stream().map(haplotype -> haplotype.toString()).sorted().collect(Collectors.toList())) {
+                    assemblyDebugOutStream.write(haplotype);
```

Comment:
Here too you've switched from calling `println()` (which writes a newline to the stream) to calling `write()` (which I don't believe writes a newline).

Reply:
Fixed.